tso-server/src/main/java/com/yahoo/omid/tso/PersistenceProcessorImpl.java

```diff
@@ -15,345 +15,216 @@
+            RetryProcessor retryProc,
+            Panicker panicker)
+                    throws InterruptedException, ExecutionException, IOException {
         this(config,
-             metrics,
-             tsoHostAndPort,
-             new Batch(config.getMaxBatchSize()),
-             leaseManager,
-             commitTable,
-             reply,
-             retryProc,
-             panicker);
-
+                metrics,
+                null,
+                tsoHostAndPort,
+                leaseManager,
+                commitTable,
+                reply,
+                retryProc,
+                panicker);
     }
 
     @VisibleForTesting
     PersistenceProcessorImpl(TSOServerConfig config,
                              MetricsRegistry metrics,
-                             String tsoHostAndPort,
                              Batch batch,
+                             String tsoHostAndPort,
                              LeaseManagement leaseManager,
                              CommitTable commitTable,
                              ReplyProcessor reply,
                              RetryProcessor retryProc,
                              Panicker panicker)
-            throws IOException {
+    throws InterruptedException, ExecutionException, IOException {
 
-        this.tsoHostAndPort = tsoHostAndPort;
-        this.batch = batch;
-        this.leaseManager = leaseManager;
         this.commitTableClient = commitTable.getClient();
         this.writer = commitTable.getWriter();
         this.reply = reply;
         this.retryProc = retryProc;
         this.panicker = panicker;
 
-        LOG.info("Creating the persist processor with batch size {}, and timeout {}ms",
-                config.getMaxBatchSize(), config.getBatchPersistTimeoutMS());
-
-        flushTimer = metrics.timer(name("tso", "persist", "flush"));
-        batchSizeHistogram = metrics.histogram(name("tso", "persist", "batchsize"));
-        timeoutMeter = metrics.meter(name("tso", "persist", "timeout"));
-
-        // FIXME consider putting something more like a phased strategy here to avoid
-        // all the syscalls
-        final TimeoutBlockingWaitStrategy timeoutStrategy
-                = new TimeoutBlockingWaitStrategy(config.getBatchPersistTimeoutMS(), TimeUnit.MILLISECONDS);
+        batchIDCnt = 0L;
 
-        persistRing = RingBuffer.createSingleProducer(
-                PersistEvent.EVENT_FACTORY, 1 << 20, timeoutStrategy); // 2^20 entries in ringbuffer
-        SequenceBarrier persistSequenceBarrier = persistRing.newBarrier();
-        BatchEventProcessor<PersistEvent> persistProcessor = new BatchEventProcessor<>(
-                persistRing,
-                persistSequenceBarrier,
-                this);
-        persistRing.addGatingSequences(persistProcessor.getSequence());
-        persistProcessor.setExceptionHandler(new FatalExceptionHandler(panicker));
+        batchPool = new BatchPool(config, batch);
```

Comment:
You probably wanted to have BatchPool instance injected rather than create it here.